R/fuzzy_match_mafs.R

```diff
@@ -0,0 +1,119 @@
+#' @title Fuzzy match mafs
+#'
+#' @description Compare two MAFs and identify pairs of rows as likely matches using approximate positions (mostly for indels)
+#'
+#' @details Specify the two mafs in a data frame format to compare as `maf1` and `maf2`. In the generated output, the data
+#' frame is derived from maf1 with two new columns (maf1_ID and maf2_ID). Values in maf1_ID are derived from the maf1.
+#' The values in maf2_ID are derived from maf2 and are NA for any rows that could not be matched to maf2.
+#'
+#' @param maf1 The index maf (akin to the first argument in a left_join). The expected input format is data frame. The minimal required columns are Tumor_Sample_Barcode, Chromosome, Start_Position, End_Position, Reference_Allele, and Tumor_Seq_Allele2.
+#' @param maf2 The maf you will search for potential matches to rows in maf1 (akin to the second argument in a left_join). Has the same minimal required columns as the maf file supplied with maf1 argument.
+#' @param padding Number of nucleotides of padding to add around Start_Position and End_Position for fuzzy match. The default is 0 (no padding).
+#' @param matching_only Set to TRUE if you only want the rows of maf1 that were successfully matched to maf2. Default is FALSE.
+#' @param invert Set to TRUE if you ony want the rows of maf1 that could not be matched to maf2. Default is FALSE.
+#'
+#' @return data frame 
+#'
+#' @rawNamespace import(data.table, except = c("last", "first", "between", "transpose"))
+#' @import dplyr tidyr tibble
+#' @export
+#'
+#' @examples
+#' \dontrun{
```

Comment:
Maybe it makes sense to add working examples and remove the `dontrun` statement. See this review for examples that are working, using the bundled data in GAMBLR.

Reply:
I've added a real example, and removed the don't run wrapping. I will update this function a bit to remove the dependency on faulty data.table and then will update this example as well